test/integration/apiserver/runtime_config_test.go

```diff
@@ -0,0 +1,170 @@
+// +build integration,!no-etcd
+
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package apiserver
+
+import (
+	"fmt"
+	"io/ioutil"
+	"net"
+	"net/http"
+	"os"
+	"testing"
+	"time"
+
+	"k8s.io/kubernetes/cmd/kube-apiserver/app"
+	"k8s.io/kubernetes/cmd/kube-apiserver/app/options"
+	"k8s.io/kubernetes/pkg/api/testapi"
+)
+
+// Starts the apiserver. It takes an index to start the apiserver on a different port each time.
+// Returns the server IP and a channel to stop it.
+func run(t *testing.T, runtimeConfig string, index int) (string, chan struct{}) {
+	certDir, err := ioutil.TempDir("", "")
+	if err != nil {
+		t.Fatalf("Failed to create temporary certificate directory: %v", err)
+	}
+	defer os.RemoveAll(certDir)
+
+	s := options.NewServerRunOptions()
+	_, serviceClusterIPRange, err := net.ParseCIDR("10.0.0.0/24")
+	if err != nil {
+		t.Fatalf("unexpected error in generation service cluster IP range: %v", err)
+	}
+	s.ServiceClusterIPRange = *serviceClusterIPRange
+	securePort := 6450 + index
```

Comment:
I think that hard-coding ports is guaranteed to be a source of flakes at some point, whether testing locally or in ci.  The fact that it hasn't been a problem thus far is not a good indication that it's a good idea.  Consider taking the approach of #42278 and #42059 and make the port selection dynamic.